R/sugar.R

```diff
@@ -20,3 +20,162 @@ tnr = function(.key, ...) {
 tnrs = function(.keys, ...) {
   dictionary_sugar_mget(mlr_tuners, .keys, ...)
 }
+
+#' @title Syntactic Sugar for Automatic Tuning
+#' 
+#' @description
+#' Function to create an [AutoTuner].
+#' 
+#' @param method (`character(1)`)\cr
+#'  Key to retrieve tuner from [mlr_tuners] dictionary.
+#' @param term_evals (`integer(1)`)\cr
+#'  Number of allowed evaluations.
+#' @param term_time (`integer(1)`)\cr
+#'  Maximum allowed time in seconds.
+#' @param ... (named `list()`)\cr
+#'  Named arguments to be set as parameters of the tuner.
+#' 
+#' @return [AutoTuner]
+#' 
+#' @template param_learner
+#' @template param_resampling
+#' @template param_measure
+#' @template param_search_space
+#' 
+#' @export
+#' @examples
+#' learner = lrn("classif.rpart")
+#' learner$param_set$values$minsplit = to_tune(1, 10)
+#'
+#' at = tune_auto(
+#'   method = "random_search",
+#'   learner = learner, 
+#'   resampling = rsmp ("holdout"),
+#'   measure = msr("classif.ce"), 
+#'   term_evals = 50, 
+#'   batch_size = 10)  
+#'
+#' at$train(tsk("pima"))
+tune_auto = function(method, learner, resampling, measure, term_evals = NULL, term_time = NULL, search_space = NULL,
```

Comment:
```suggestion
auto_tuner = function(method, learner, resampling, measure, term_evals = NULL, term_time = NULL, search_space = NULL,
```
`tune_auto` sounds like this function actually tunes something. The other sugar functions are also called like the object classes they create. Actually `ttnr()` would be consequent :smiley: or `attnr()`?

Reply:
- Maybe it makes sense to allow `measure()` to be set automatically?
- Do we really want to disallow to pass the `search_space` directly?